### PR TITLE
Revert Add Multi-Currency Support to Page Caching via Cookies

### LINKED ISSUE #8679: Site Health Reports Performance errors related to wcpay_currency cookie
### Describe the bug

Related to #8534

After updating to WooPayments 7.5.1, Site Health sometimes reports a critical performance issue. 

```
Page cache is not detected and the server response time is slow

Page cache enhances the speed and performance of your site by saving and serving static pages instead of calling for a page every time a user visits.

Page cache is detected by looking for an active page cache plugin as well as making three requests to the homepage and looking for one or more of the following HTTP client caching response headers:

cache-control, expires, age, last-modified, etag, x-cache-enabled, x-cache-disabled, x-srcache-store-status, x-srcache-fetch-status.
 Median server response time was 1,419 milliseconds. It should be less than the recommended 600 milliseconds threshold.
 No client caching response headers were detected.
 A page cache plugin was not detected.
[Learn more about page cache(opens in a new tab)](https://wordpress.org/documentation/article/optimization/#Caching)
```

Upon checking in hosting, it appeared to be related to this string:
```
set-cookie: wcpay_currency=GBP_1; expires=Fri, 19-Apr-2024 11:05:09 GMT; Max-Age=3600; path=/
```

### To Reproduce

1. Have WooPayments 7.5.1 installed with multicurrency enabled
2. Go to Tools > Site Health
3. See the error once Site Health completes loading 

### Actual behavior

The error seems to be temporary in some cases, I reproduced it with Multi-currency enabled but no additional currencies as well as multiple added currencies.  
The merchant did not notice any change in site behavior yet, but was concerned about the critical warning. 

### Screenshots

Screenshot 2024-04-19 at 5 01 03 PM

### Expected behavior

No performance change with the additions of  #8645

### Additional context

8070554-zd-a8c

### PR DESCRIPTION
**Pull request by ricardo**
#### Changes proposed in this Pull Request

- Revert Automattic/woocommerce-payments#8534
- Fixes #8679 

#### Testing instructions

**Prerequisites**

- Switch to this branch.
- Enable the multi-currency feature in WooPayments settings and configure a few different currencies.

**Test 1: Ensure the `wcpay_currency` cookie is no longer set**

1. Clear your browser cookies to ensure a fresh start for this test.
2. Visit your site and switch to a different currency using the currency switcher provided by WooPayments.
3. After switching currencies, check your browser's developer tools under the Application > Storage tab and ensure there is no `wcpay_currency` cookie.

**Test 2: Ensure product pages are served from Batcache**

1. Clear your browser cookies to ensure a fresh start for this test.
2. Follow the instructions on how to set up Batcache on your local environment: pcrvl0-1cG-p2
3. Visit a product page and refresh it once.
4. Ensure the page is served from Batcache by inspecting the page source code.

**Note:** It's expected that the page is not served from cache if there's an active WP session, or if you have switched the store currency. To ensure a clear state for this test, clear your browser cookies.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8738` or branch name `revert-8534-8130/add-cookie-product-cache-per-currency` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=revert-8534-8130/add-cookie-product-cache-per-currency) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 9ee85082526011c14f06240cf214b4245c9ddf36
- Build time: 2024-05-01 16:38:52 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: logstash.log
Comment (reykjalin) [inline logstash.log:1]: I don't think this file is supposed to be here, so I suggest we remove this? 🤔

**Review by reykjalin — CHANGES_REQUESTED** [review]
The code changes look good aside from a log file I don't think you meant to commit.

I wasn't able to follow the testing instructions because I ran into issues setting up memcached and batcache. We can continue to discuss that in pcrvl0-1cG-p2#comment-604

File: logstash.log
Comment (ricardo) [reply logstash.log:1]: I might have added it accidentally when committing the changelog entry.

**Review by reykjalin — APPROVED** [review]
Looks good and tests well 🎉 

I had to make sure the PRBs were not shown on product pages to make sure there was no WC session created, and after doing that the test worked as expected. Batcache is able to cache both the front page and product pages 🙌

**Pull request merged**